src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -190,6 +190,25 @@ public async Task StopAsync(CancellationToken cancellationToken)
         {
             _logger.LogDebug(ex, "One or more monitoring tasks terminated with an error.");
         }
+
+        try
+        {
+            // The app orchestrator (represented by kubernetesService here) will perform a resource cleanup
+            // (if not done already) when the app host process exits.
+            // This is just a perf optimization, so we do not care that much if this call fails.
+            // There is not much difference for single app run, but for tests that tend to launch multiple instances
+            // of app host from the same process, the gain from programmatic orchestrator shutdown is significant
+            // See https://github.com/dotnet/aspire/issues/6561 for more info.
+            await kubernetesService.StopServerAsync(Model.ResourceCleanup.Full, cancellationToken).ConfigureAwait(false);
```

Comment:
Does this block?

Reply:
It starts the cleanup process but does not wait for it to finish. So no.